### ISSUE TITLE
wallet tool silently ignores trailing options

### ISSUE THREAD

**Issue by MarcoFalke** [bitcoin/bitcoin#20902]
Steps to reproduce:

```
$ ./src/bitcoin-wallet -datadir=/tmp  info  -wallet=asdf
Wallet info
===========
Name: 
Format: bdb
Descriptors: no
Encrypted: no
HD (hd seed available): yes
Keypool Size: 2000
Transactions: 0
Address Book: 0
```

(See the mismatching wallet name!)

Compare to:

```
$ ./src/bitcoin-wallet -datadir=/tmp  -wallet=asdf info
Wallet info
===========
Name: asdf
Format: bdb
Descriptors: no
Encrypted: no
HD (hd seed available): yes
Keypool Size: 2000
Transactions: 0
Address Book: 0

**Comment by MarcoFalke** [comment]
Fixed in #20715

**Comment by samsungbiologics** [comment]
bug and fix ? hahaa
